File: android/src/main/kotlin/com/virtusize/virtusize_flutter_sdk/VirtusizeFlutterConstants.kt
internal object VirtusizeFlutterMethod {
    // Flutter to Android
    const val SET_VIRTUSIZE_PARAMS = "setVirtusizeParams"
    const val SET_USER_ID = "setUserId"
    const val OPEN_VIRTUSIZE_WEB_VIEW = "openVirtusizeWebView"
    const val GET_PRIVACY_POLICY_LINK = "getPrivacyPolicyLink"
    const val SEND_ORDER = "sendOrder"
    const val LOAD_VIRTUSIZE = "loadVirtusize"
    const val SHOW_PRIVACY_POLICY = "showPrivacyPolicy"

    // Android to Flutter
    const val ON_VS_EVENT = "onVSEvent"
    const val ON_VS_ERROR = "onVSError"
    const val ON_PRODUCT = "onProduct"
    const val ON_PRODUCT_DATA_CHECK = "onProductDataCheck"
    const val ON_REC_CHANGE = "onRecChange"
    const val ON_LANGUAGE_CLICK = "onLanguageClick"
    const val ON_PRODUCT_ERROR = "onProductError"
}

internal object VirtusizeFlutterKey {
    const val API_KEY = "apiKey"
    const val EXTERNAL_USER_ID = "externalUserId"
    const val ENVIRONMENT = "env"
    const val LANGUAGE = "language"
    const val SHOW_SGI = "showSGI"
    const val ALLOW_LANGUAGES = "allowedLanguages"
    const val DETAILS_PANEL_CARDS = "detailsPanelCards"
    const val SHOW_SNS_BUTTONS = "showSNSButtons"
    const val BRANCH = "branch"
    const val SERVICE_ENVIRONMENT = "serviceEnvironment"
    const val EXTERNAL_PRODUCT_ID = "externalProductId"
    const val IMAGE_URL = "imageURL"
    const val CLOUDINARY_IMAGE_URL = "cloudinaryImageURL"
    const val STORE_PRODUCT_ID = "storeProductId"
    const val VIRTUSIZE_PARAMS = "virtusizeParams"
    const val DISPLAY_LANGUAGE = "displayLanguage"
    const val IMAGE_TYPE = "imageType"
    const val PRODUCT_TYPE = "productType"
    const val PRODUCT_STYLE = "productStyle"
    const val REC_TEXT = "recText"
    const val SHOW_USER_PRODUCT_IMAGE = "showUserProductImage"
    const val IS_VALID_PRODUCT = "isValidProduct"
    const val SHOW_PRIVACY_POLICY = "showPrivacyPolicy"
    const val STORE_NAME = "storeName"
}

internal object VirtusizeEventKey {
    const val SHORT_EVENT_NAME = "name"
    const val EVENT_NAME = "eventName"
    const val USER_PRODUCT_ID = "userProductId"
    const val REC_TYPE = "recommendationType"
    const val SIZE_REC_NAME = "sizeRecName"
}

File: ios/Classes/VirtusizeFlutterConstants.swift
struct VirtusizeFlutterMethod {
	/// Flutter to iOS
	static let setVirtusizeParams = "setVirtusizeParams"
	static let setUserId = "setUserId"
	static let openVirtusizeWebView = "openVirtusizeWebView"
	static let getPrivacyPolicyLink = "getPrivacyPolicyLink"
	static let sendOrder = "sendOrder"
	static let loadVirtusize = "loadVirtusize"
	
	/// iOS to Flutter
	static let onVSEvent = "onVSEvent"
	static let onVSError = "onVSError"
	static let onProduct = "onProduct"
	static let onProductDataCheck = "onProductDataCheck"
	static let onRecChange = "onRecChange"
	static let onProductError = "onProductError"
}

struct VirtusizeFlutterKey {
	static let apiKey = "apiKey"
	static let externalUserId = "externalUserId"
	static let environment = "env"
	static let language = "language"
	static let showSGI = "showSGI"
	static let allowedLanguages = "allowedLanguages"
	static let detailsPanelCards = "detailsPanelCards"
	static let showSNSButtons = "showSNSButtons"
	static let branch = "branch"
	static let externalProductId = "externalProductId"
	static let imageURL = "imageURL"
	static let cloudinaryImageURL = "cloudinaryImageURL"
	static let storeProductId = "storeProductId"
	static let virtusizeParams = "virtusizeParams"
	static let displayLanguage = "displayLanguage"
	static let imageType = "imageType"
	static let productType = "productType"
	static let productStyle = "productStyle"
	static let recText = "recText"
	static let showUserProductImage = "showUserProductImage"
	static let isValidProduct = "isValidProduct"
}

struct VirtusizeEventKey {
	static let shortEventName = "name"
	static let eventName = "eventName"
	static let userProductID = "userProductId"
	static let recType = "recommendationType"
	static let sizeRecName = "sizeRecName"
}
